v8context.cc

```diff
@@ -176,20 +180,23 @@ PersistentValuePtr V8Context::Eval(const char* source, const char* filename) {
   try_catch.SetVerbose(false);
 
   mLastError.clear();
+  mTerminated = false;
 
   v8::Local<v8::Script> script = v8::Script::Compile(
       v8::String::NewFromUtf8(mIsolate, source),
       filename ? v8::String::NewFromUtf8(mIsolate, filename)
                : v8::String::NewFromUtf8(mIsolate, "undefined"));
 
   if (script.IsEmpty()) {
+    mTerminated = try_catch.HasTerminated();
```

Comment:
We will want to set both mTerminated and mLastError on every exit, right? Is it worth ensuring correct cleanup by writing a little utility class that takes a bool*, a std::string*, and a v8::TryCatch*, and updates the first two from HasTerminated in its destructor? I think this would also be used in Apply.